source/applications/initializers.md

```diff
@@ -0,0 +1,100 @@
+Initializers provide an opportunity to configure your application as it boots.
+
+There are two types of initializers: application initializers and application
+instance initializers.
+
+Application initializers are run as your application boots, and provide the
+primary means to configure [dependency injections](./dependency-injection) in
+your application.
+
+Application instance initializers are run as an application instance is loaded.
+They provide a way to configure the initial state of your application, as well
+as to set up dependency injections that are local to the application instance
+(e.g. A/B testing confurations).
+
+Initializers can perform any configuration operations, as long as they are
+synchronous. The user experience is degraded by attempting to block the booting
```

Comment:
Well, that sounds like a very good example of one of those "slim" use cases, which may not be as rare as I think!

Here's an attempt to straddle this issue by heavily discouraging asynchrony, but recognizing it as a possibility:

> Operations performed in initializers should be kept as lightweight as possible
to minimize delays in loading your application. Although advanced techniques
exist for allowing asynchrony in application initializers (i.e. `deferReadiness` 
and `advanceReadiness`), these techniques should generally be avoided. Any
asynchronous loading conditions (e.g. user authorization) are almost always
better handled in your application route's hooks, which allows for DOM
interaction while waiting for conditions to resolve.

If that works for you, I'll update again.